File: assertion-roulette/python-pytest/CrudAppServiceTests_Should_Create_Async.py
import pytest

@pytest.mark.asyncio
async def test_should_create_async(product_crud_app_service, default_test_db_context, get_default_test_db_context):
    user_output = await product_crud_app_service.create_async({"Code": "create_async_product_code", "Name": "Create Async Product Name"})
    await default_test_db_context.save_changes_async()
    another_scope_db_context = get_default_test_db_context()
    inserted_product_dto = await another_scope_db_context.products.find_async(user_output["Id"])
    assert user_output is not None
    assert inserted_product_dto is not None
    assert user_output["Code"] == inserted_product_dto["Code"]

File: assertion-roulette/CSharpXUnit/CSharpXUnit.Tests/Tests/CrudAppServiceTests_Should_Create_Async.cs
using Xunit;

namespace BackOfficeBase.Tests.Application
{
    public class CrudAppServiceTests
    {
        [Fact]
        public void Should_Create_Async()
        {
            {
                var userOutput = await _productCrudAppService.CreateAsync(new CreateProductInput { Code = "create_async_product_code", Name = "Create Async Product Name" });
                await DefaultTestDbContext.SaveChangesAsync();
                var anotherScopeDbContext = GetDefaultTestDbContext();
                var insertedProductDto = await anotherScopeDbContext.Products.FindAsync(userOutput.Id);
                Assert.NotNull(userOutput);
                Assert.NotNull(insertedProductDto);
                Assert.Equal(userOutput.Code, insertedProductDto.Code);
            }
        }
    }
}
